### PR TITLE
CLDCONNECT-2407: Refactored according to Devkit v3.7.0 guidelines

### PR DESCRIPTION
**Pull request by karthikanantha**
Note: Added templates for release-notes.adoc and user-manual.adoc for the time being in-order for project structure validation to pass. Will be updated in the doc phase.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -13,7 +7,14 @@
...
     <name>Mule CMIS Connector</name>
 
+    <parent>
+        <groupId>org.mule.tools.devkit</groupId>
+        <artifactId>mule-devkit-parent</artifactId>
+        <version>3.7.0</version>
+    </parent>
+
     <properties>
+        <jdk.version>1.6</jdk.version>
```
Comment (tejaswip) [inline pom.xml]: Is there a specific reason for using JDK 1.6 ?

Comment (karthikanantha) [inline pom.xml]: To support the connector on mule 3.5.x running with JDK 1.6

File: src/main/java/org/mule/module/cmis/CMISConnector.java [hunk 2]
```diff
@@ -687,10 +565,10 @@ public void delete(@Default("#[payload]") CmisObject cmisObject,
     @Processor
     public List<String> deleteTree(@Placement(order = 1) @Default("#[payload]") CmisObject folder,
                                    @Placement(order = 2) @Optional String folderId,
-                                   @Placement(order = 4) boolean allversions,
                                    @Placement(order = 3) @Optional UnfileObject unfile,
+                                   @Placement(order = 4) boolean allversions,
                                    @Placement(order = 5) boolean continueOnFailure) {
-        return facade.deleteTree(folder, folderId, allversions, unfile, continueOnFailure);
+        return facade.deleteTree(folder, folderId, unfile, allversions, continueOnFailure);
```
Comment (tejaswip) [inline src/main/java/org/mule/module/cmis/CMISConnector.java]: Please update the order of parameters in Java Doc appropriately.

Comment (karthikanantha) [inline src/main/java/org/mule/module/cmis/CMISConnector.java]: Updated the PR with the suggested java doc changes.

**Pull request merged**